src/libraries/System.Security.Cryptography.Pkcs/src/System/Security/Cryptography/Pkcs/SignerInfo.cs

```diff
@@ -706,7 +706,7 @@ private void Verify(
 
             if (!verifySignatureOnly)
             {
-                X509Chain chain = new X509Chain();
+                using X509Chain chain = new X509Chain();
```

Comment:
Instead of the implicit scope using, this should be a normal block-body using statement, and the end of the block should be at the end of the `if` that calls `chain.Build()` (so between that block and the EKU check)